pkg/varlog/options.go

```diff
@@ -182,6 +185,12 @@ func withoutSelectLogStream() AppendOption {
 	})
 }
 
+func WithAllowLogStreams(logStreams map[types.LogStreamID]struct{}) AppendOption {
+	return newAppendOption(func(opts *appendOptions) {
+		opts.allowLogStreams = logStreams
+	})
```

Comment:
Is it necessary to handle an empty map passed by users?

Reply:
This prevents appending from the zombie LogStream, even if the user accidentally sends an empty list.

Reply:
If a user sets an empty map, will the API not try to append logs at all?
Let me give you an example code:

```go
allowed := map[types.LogStreamID]struct{}{}
// It never append.
vlog.Append(ctx, tpid, [][]byte{[]byte("hello"), []byte("varlog")}, vlog.WithAllowedLogStreams(allowed))
```

Reply:
Right. The purpose of this is to avoid append to unmanaged LogStream. If the user's filter logic is wrong and an empty allowed is passed, it's better to explicitly throw an error. If allowed = nil, it will be append to any LogStream.

Reply:
I got it!